File: interviewplannerapp_ui/src/services/ApplicationFormService.js
import http from "../http-common"; 

class ApplicationFormService {
  getAllApplicationForms(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/applicationForm/applicationForms`, searchDTO);
  }

  get(applicationFormId) {
    return this.getRequest(`/applicationForm/${applicationFormId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/applicationForm?field=${matchData}`, null);
  }

  addApplicationForm(data) {
    return http.post("/applicationForm/addApplicationForm", data);
  }

  update(data) {
  	return http.post("/applicationForm/updateApplicationForm", data);
  }
  
  uploadImage(data,applicationFormId) {
  	return http.postForm("/applicationForm/uploadImage/"+applicationFormId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ApplicationFormService();

File: src/main/java/com/interviewplannerapp/service/ApplicationFormService.java
package com.interviewplannerapp.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.interviewplannerapp.domain.ApplicationForm;
import com.interviewplannerapp.dto.ApplicationFormDTO;
import com.interviewplannerapp.dto.ApplicationFormSearchDTO;
import com.interviewplannerapp.dto.ApplicationFormPageDTO;
import com.interviewplannerapp.dto.ApplicationFormConvertCriteriaDTO;
import com.interviewplannerapp.service.GenericService;
import com.interviewplannerapp.dto.common.RequestDTO;
import com.interviewplannerapp.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ApplicationFormService extends GenericService<ApplicationForm, Integer> {

	List<ApplicationForm> findAll();

	ResultDTO addApplicationForm(ApplicationFormDTO applicationFormDTO, RequestDTO requestDTO);

	ResultDTO updateApplicationForm(ApplicationFormDTO applicationFormDTO, RequestDTO requestDTO);

    Page<ApplicationForm> getAllApplicationForms(Pageable pageable);

    Page<ApplicationForm> getAllApplicationForms(Specification<ApplicationForm> spec, Pageable pageable);

	ResponseEntity<ApplicationFormPageDTO> getApplicationForms(ApplicationFormSearchDTO applicationFormSearchDTO);
	
	List<ApplicationFormDTO> convertApplicationFormsToApplicationFormDTOs(List<ApplicationForm> applicationForms, ApplicationFormConvertCriteriaDTO convertCriteria);

	ApplicationFormDTO getApplicationFormDTOById(Integer applicationFormId);







}





